app/assets/javascripts/admin/orders/controllers/orders_controller.js.coffee

```diff
@@ -23,10 +23,15 @@ angular.module("admin.orders").controller "ordersCtrl", ($scope, $timeout, Reque
     $scope.fetchResults()
 
   $scope.fetchResults = (page=1) ->
+    startDate = $scope['q']['completed_at_gteq']
+    $scope.formattedStartDate = if startDate then moment(startDate).startOf('day').format() else null
+    endDate = $scope['q']['completed_at_lteq']
+    $scope.formattedEndDate = if endDate then moment(endDate).endOf('day').format() else null
```

Comment:
1. have you tried to just use completed_at_lteq with $scope['q']['completed_at_gteq'] ? I thought it could work just with the date.
2. do you really need to usse $scope for $scope.formattedEndDate, wouldnt simply formattedEndDate work?
3. assigments with conditionals and assigning to null are 2 things we should try to avoid, is there a way we can do this without those two things?
Removing $scope and assigment with conditional
```
endDate = $scope['q']['completed_at_lteq']
formattedEndDate = null
formattedEndDate = moment(endDate).endOf('day').format() if endDate
```
and maybe you can just remove the line "formattedEndDate = null" because undefined on formattedEndDate will just work when set on the params?